docs: cqn4sql path expression transformations (#1558)
cap-js/cds-dbs: db-service/doc/cqn4sql-transformations.md

```sql
-- Input
SELECT from Books { genre.parent.descr }

-- Output
SELECT from Books
  left outer join Genres as genre   on genre.ID = Books.genre_ID
  left outer join Genres as parent  on parent.ID = genre.parent_ID
{ parent.descr as genre_parent_descr }
```

### Structures within association paths

Structure steps are absorbed into the flattened column name — only association steps produce joins:

```sql
-- Input
SELECT from Books { dedication.addressee.name }
-- `dedication` is a structure, `addressee` is an association inside it

-- Output
SELECT from Books
  left outer join Person as addressee
    on addressee.ID = Books.dedication_addressee_ID
{ addressee.name as dedication_addressee_name }
```

### Paths in WHERE / HAVING / ORDER BY

Association paths in query modifiers produce the same joins:

```sql
-- Input
SELECT from Books { ID } where author.name = 'Brontë'

-- Output
SELECT from Books
  left outer join Authors as author on author.ID = Books.author_ID
{ Books.ID }
where author.name = 'Brontë'
```

If a column and a WHERE clause both traverse the same association, they share the join.

### FK shortcut — no join needed

> If the path accesses only a foreign key field (or the association itself without further navigation), no join is generated:

Reviewer: clarify: what if also another field is accessed via the assoc?

```sql
-- Input
SELECT from Books { author.ID }

-- Output (no join)
SELECT from Books { Books.author_ID as author_ID }

-- Selecting the association itself replaces it with its FK(s)
SELECT from Books { author }
-- Output
SELECT from Books { Books.author_ID }
```

---

## 2. Infix Filters → Additional JOIN ON Conditions

An infix filter on an association step adds the filter predicate to the join's `ON` clause:

```sql
-- Input
SELECT from Books { author[placeOfBirth = 'Marbach'].name }

-- Output
SELECT from Books
  left outer join Authors as author
    on author.ID = Books.author_ID
    and author.placeOfBirth = 'Marbach'
{ author.name as author_name }
```

> The same association with **different filters** (or with/without a filter) produces **separate joins** with unique aliases:

Reviewer: why "with/without"?

Author: → the same association once with a filter and once without a filter produce separate joins

```sql
-- Input
SELECT from Books {
  author[placeOfBirth = 'Marbach'].name as n1,
  author.name as n2
}

-- Output
SELECT from Books
  left outer join Authors as author
    on author.ID = Books.author_ID and author.placeOfBirth = 'Marbach'
  left outer join Authors as author2
    on author2.ID = Books.author_ID
{
  author.name  as n1,
  author2.name as n2
}
```

> Infix filters also apply in `exists` and `FROM` navigation, where they become conditions in the `WHERE EXISTS` subquery.

Reviewer: ```suggestion
Infix filters also can occur in `exists` and `FROM` navigation, where they become conditions in the `WHERE EXISTS` subquery.
```

---

## 3. WHERE EXISTS Subqueries

> Both association paths in `FROM` and the `exists` predicate produce `WHERE EXISTS` correlated subqueries. The mechanics are the same — a `FROM` path like `Authors:books` is semantically equivalent to `SELECT from Books where exists author`.

Reviewer: ```suggestion
Association paths both in `FROM` and in an `exists` predicate produce `WHERE EXISTS` correlated subqueries. The mechanics are the same — a `FROM` path like `Authors:books` is semantically equivalent to `SELECT from Books where exists author`.
```

### Association in FROM

> An association step in the `FROM` clause rewrites the query target to the association's target entity and adds a `WHERE EXISTS` subquery correlating back to the source:

Reviewer: ```suggestion
An association step in the `FROM` clause moves the query target to the association's target entity and adds a `WHERE EXISTS` subquery correlating back to the source:
```
